exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2606 @@
+	if err != nil {
+		t.Error("InitiateIsolatedMarginQuickRepayment() error", err)
+	}
+}
+
+func TestInitiateIsolatedMarginSingleRepayment(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(cantManipulateRealOrdersOrKeysNotSet)
+	}
+
+	err := ku.InitiateIsolatedMarginSingleRepayment(context.Background(), "BTC-USDT", "USDT", "628c570f7818320001d52b69", 10)
+	if err != nil {
+		t.Error("InitiateIsolatedMarginSingleRepayment() error", err)
+	}
+}
+
+func TestGetCurrentServerTime(t *testing.T) {
+	t.Parallel()
+
+	_, err := ku.GetCurrentServerTime(context.Background())
+	if err != nil {
+		t.Error("GetCurrentServerTime() error", err)
+	}
+}
+
+func TestGetServiceStatus(t *testing.T) {
+	t.Parallel()
+	_, err := ku.GetServiceStatus(context.Background())
+	if err != nil {
+		t.Error("GetServiceStatus() error", err)
+	}
+}
+
+func TestPostOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(cantManipulateRealOrdersOrKeysNotSet)
+	}
+
+	// default order type is limit
+	_, err := ku.PostOrder(context.Background(), "5bd6e9286d99522a52e458de", "buy", "BTC-USDT", "limit", "", "", "", 0.1, 0, 0, 0, 0, true, false, false)
+	if err != nil {
+		t.Error("PostOrder() error", err)
+	}
+
+	// market order
+	_, err = ku.PostOrder(context.Background(), "5bd6e9286d99522a52e458de", "buy", "BTC-USDT", "market", "remark", "", "", 0.1, 0, 0, 0, 0, true, false, false)
+	if err != nil {
+		t.Error("PostOrder() error", err)
+	}
+}
+
+func TestPostMarginOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(cantManipulateRealOrdersOrKeysNotSet)
+	}
+
+	// default order type is limit and margin mode is cross
+	_, err := ku.PostMarginOrder(context.Background(), "5bd6e9286d99522a52e458de", "buy", "BTC-USDT", "", "", "", "", "10000", 1000, 0.1, 0, 0, 0, true, false, false, false)
+	if err != nil {
+		t.Error("PostMarginOrder() error", err)
+	}
+
+	// market isolated order
+	_, err = ku.PostMarginOrder(context.Background(), "5bd6e9286d99522a52e458de", "buy", "BTC-USDT", "market", "remark", "", "isolated", "", 1000, 0.1, 0, 0, 5, true, false, false, true)
+	if err != nil {
+		t.Error("PostMarginOrder() error", err)
+	}
+}
+
+func TestPostBulkOrder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(cantManipulateRealOrdersOrKeysNotSet)
+	}
+
+	req := []OrderRequest{
+		{
+			ClientOID: "3d07008668054da6b3cb12e432c2b13a",
+			Side:      "buy",
+			Type:      "limit",
+			Price:     1000,
+			Size:      0.01,
+		},
+		{
+			ClientOID: "37245dbe6e134b5c97732bfb36cd4a9d",
+			Side:      "buy",
+			Type:      "limit",
+			Price:     1000,
+			Size:      0.01,
+		},
+	}
+
+	_, err := ku.PostBulkOrder(context.Background(), "BTC-USDT", req)
+	if err != nil {
+		t.Error("PostBulkOrder() error", err)
```

Comment:
PostBulkOrder() error json: cannot unmarshal object into Go struct field Response.data of type []kucoin.PostBulkOrderResp